packages/mudtokens/src/erc20/LibERC20.sol

```diff
@@ -3,13 +3,20 @@ pragma solidity >=0.8.0;
 
 import { IBaseWorld } from "@latticexyz/world/src/interfaces/IBaseWorld.sol";
 import { ResourceSelector, ROOT_NAMESPACE } from "@latticexyz/world/src/ResourceSelector.sol";
+import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
 import { AllowanceTable, BalanceTable, MetadataTable } from "../codegen/Tables.sol";
 import { ERC20_ALLOWANCE_T as ALLOWANCE, ERC20_BALANCE_T as BALANCE, ERC20_METADATA_T as METADATA } from "../common/constants.sol";
 import { IERC20 } from "./interfaces/IERC20.sol";
 import { IERC20Metadata } from "./interfaces/IERC20Metadata.sol";
 import { ERC20Proxy } from "./ERC20Proxy.sol";
 
 library LibERC20 {
+  modifier onlyProxyWorld(bytes16 namespace) {
+    ERC20Proxy _proxy = ERC20Proxy(proxy(namespace));
+    require(address(_proxy.world()) == StoreSwitch.getStoreAddress(), "ERC20: invalid world");
```

Comment:
We need to check system and world because in the proxy, the msg.sender will be the World if its being called from the root namespace while non-root namespaces will be the System. Does this answer your question?

Reply:
My understanding is that `StoreSwitch.getStoreAddress()` will always return `msg.sender` because I didn't see `StoreSwitch.setStoreAddress()` anywhere. So I read this as the proxy's world must equal to `msg.sender`, which means `msg.sender` is always a world.

If the world address is defined with `StoreSwitch.setStoreAddress()`, this would read as the defined world's proxy's world (because `proxy(namespace)` would get the proxy address from the defined world's Metadata table) must equal to the defined world, which is always true (for the intended use case).